src/utils.js

```diff
@@ -0,0 +1,35 @@
+import request from 'request-promise';
+import AWS from 'aws-sdk';
+
+import logger from './logger';
+
+const s3 = new AWS.S3();
+
+export const isObjectEmpty = object => object.size === 0;
+
+export const getCompanySlug = object => object.key.split('/')[0];
+
+export const getSrcKey = object => object.key.replace(/\+/g, ' ');
```

Comment:
So, this is interesting. We call this `getSrcKey`, but it's not just getting the key, it's also converting `+` to spaces. Is that expected/appropriate? Do we _always_ want to use this decoded version whenever we request the key? Should we explicitly capture that behavior in the function name (e.g., `getDecodedSrcKey()`)?

Reply:
![image](https://user-images.githubusercontent.com/208884/60990586-c159b300-a316-11e9-9a74-da8b68ca5bf9.png)

Reply:
(To be clear, I don't know the answer to basically any of those questions, such as whether we want/expect to be dealing with the decoded key name or not. Just noting it because it made me :squint:.) (Maybe @joeltg can weigh in on this? Or maybe you already know!)

Reply:
Our code here only uses the "decoded" version -- I assume that the source / destination keys are URL encoded but that both the decoded and url encoded versions can be called "source keys" (e.g. we don't encode them at any point so... AWS must be smart enough to do that for us).

I think instead of changing the name of the method we should just make it clear in the method documentation that it returns a decoded version.  In fact, if we want to be really special we should use `decodeURI`

Reply:
Right; that's sort of where I'm circling toward. If we expect to receive an encoded version and only deal with/manipulate, then we should note that and also _probably_ use a fuller-featured decoder than this regex. (Unless that's truly the only encoded character, which may be true but probably isn't.)

Reply:
I'm having a bunch of trouble finding explicit documentation of encoding rules around S3 keys but it sure seems like the library encodes it for you before sending to amazon.  As a result we *should* have it self documented in the method name after all, since a sourceKey should be assumed to be URL encoded unless that manipulation has explicitly taken place.